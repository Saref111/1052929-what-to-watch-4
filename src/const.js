const CARD_MENU = [`Overview`, `Details`, `Reviews`];

const MONTHS = [`January`, `February`, `March`, `April`, `May`, `June`, `July`, `August`, `September`, `October`, `November`, `December`];

const GENRES = [
  `All genres`,
  `Comedy`,
  `Crime`,
  `Documentary`,
  `Drama`,
  `Horror`,
  `Kids & Family`,
  `Romance`,
  `Sci-fi`,
  `Thriller`,
];

const RATINGS = [`bad`, `normal`, `good`, `very good`, `awesome`];

const TABS = {
  OVERVIEW: 0,
  DETAILS: 1,
  REVIEWS: 2,
};

const FILMS_COUNT = 8;

const Authorization = {
  NO_AUTH: `NO_AUTH`,
  AUTH: `AUTH`,
};

export {GENRES, RATINGS, CARD_MENU, MONTHS, TABS, FILMS_COUNT, Authorization};
