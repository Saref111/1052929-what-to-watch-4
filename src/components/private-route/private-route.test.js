import React from "react";
import renderer from "react-test-renderer";
import PrivateRoute from "./private-route.jsx";
import NewReviewPage from "../new-review-page/new-review-page.jsx";
import {Routes} from "../../const.js";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import {BrowserRouter} from "react-router-dom";

const mockStore = configureStore([]);

describe(`PrivateRoute snaptest`, () => {
  test(`shouldn't render NewReviewPage`, () => {
    const store = mockStore({DATA: {
    },
    MOVIE: {
    },
    USER: {
      authorizationStatus: `NO_AUTH`,
      isSigningIn: false
    }});
    const tree = renderer.create(<Provider store={store}><BrowserRouter><PrivateRoute exact path={Routes.REVIEW}
      render={(props) => {
        return <NewReviewPage {...props}/>;
      }}
    /></BrowserRouter></Provider>).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test(`should render NewReviewPage`, () => {
    const store = mockStore({DATA: {
    },
    MOVIE: {
    },
    USER: {
      authorizationStatus: `AUTH`,
      isSigningIn: false
    }});
    const tree = renderer.create(<Provider store={store}><BrowserRouter><PrivateRoute exact path={Routes.REVIEW}
      render={(props) => {
        return <NewReviewPage {...props}/>;
      }}
    /></BrowserRouter></Provider>).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
