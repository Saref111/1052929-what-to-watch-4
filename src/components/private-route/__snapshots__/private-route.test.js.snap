// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute snaptest should render NewReviewPage 1`] = `null`;

exports[`PrivateRoute snaptest shouldn't render NewReviewPage 1`] = `null`;
